deepmd/descriptor/se_a_mask.py

```diff
@@ -0,0 +1,436 @@
+        self.type_one_side = type_one_side
+        # descrpt config. Not used in se_a_mask
+        self.sel_r = [0 for ii in range(len(self.sel_a))]
+        self.ntypes = len(self.sel_a)
+        assert self.ntypes == len(self.sel_r)
+        self.rcut_a = -1
+        # numb of neighbors and numb of descrptors
+        self.nnei_a = np.cumsum(self.sel_a)[-1]
+        self.nnei = self.nnei_a
+
+        self.ndescrpt_a = self.nnei_a * 4
+        self.ndescrpt = self.ndescrpt_a
+        self.useBN = False
+        self.dstd = None
+        self.davg = None
+        self.rcut = -1.0  # Not used in se_a_mask
+        self.compress = False
+        self.embedding_net_variables = None
+        self.mixed_prec = None
+        self.place_holders = {}
+        nei_type = np.array([])
+        for ii in range(self.ntypes):
+            nei_type = np.append(nei_type, ii * np.ones(self.sel_a[ii]))  # like a mask
+        self.nei_type = tf.constant(nei_type, dtype=tf.int32)
+
+        avg_zero = np.zeros([self.ntypes, self.ndescrpt]).astype(
+            GLOBAL_NP_FLOAT_PRECISION
+        )
+        std_ones = np.ones([self.ntypes, self.ndescrpt]).astype(
+            GLOBAL_NP_FLOAT_PRECISION
+        )
+        sub_graph = tf.Graph()
+        with sub_graph.as_default():
+            name_pfx = "d_sea_mask_"
+            for ii in ["coord", "box"]:
+                self.place_holders[ii] = tf.placeholder(
+                    GLOBAL_NP_FLOAT_PRECISION, [None, None], name=name_pfx + "t_" + ii
+                )
+            self.place_holders["type"] = tf.placeholder(
+                tf.int32, [None, None], name=name_pfx + "t_type"
+            )
+            self.place_holders["mask"] = tf.placeholder(
+                tf.int32, [None, None], name=name_pfx + "t_aparam"
+            )  # named aparam for inference compatibility in c++ interface.
+
+            self.place_holders["natoms_vec"] = tf.placeholder(
+                tf.int32, [self.ntypes + 2], name=name_pfx + "t_natoms"
+            )  # Not used in se_a_mask. For compatibility with c++ interface.
+            self.place_holders["default_mesh"] = tf.placeholder(
+                tf.int32, [None], name=name_pfx + "t_mesh"
+            )  # Not used in se_a_mask. For compatibility with c++ interface.
+
+            self.stat_descrpt, descrpt_deriv, rij, nlist = op_module.descrpt_se_a_mask(
+                self.place_holders["coord"],
+                self.place_holders["type"],
+                self.place_holders["mask"],
+                self.place_holders["box"],
+                self.place_holders["natoms_vec"],
+                self.place_holders["default_mesh"],
+            )
+        self.sub_sess = tf.Session(graph=sub_graph, config=default_tf_session_config)
+        self.original_sel = None
+
+    def get_rcut(self) -> float:
+        """
+        Returns: the cutoff radius:
+        """
+        warnings.warn("The cutoff radius is not used for this descriptor")
+        return -1.0
```

Comment:
I remember that `rcut` will be saved into the model and loaded during the MD simulations. Will `-1` work? I think it should be `999.0` or `float("inf")` (not sure if inf works) here.

Reply:
I did test the c++ interface for frozen model with this op. It works.